### ISSUE TITLE
Can't run a Cilicon via Gitlab runner (Citadel.SSHClient.CommandFailed error 1.)

### ISSUE THREAD

**Issue by kpolupanova** [traderepublic/Cilicon#55]
Hi! I've tried to start workin with cilicon and get an error that I can't figure out how to fix.
After launc Cilicon i get this error:
**Configuring GitLab Runner...
The operation couldn’t be completed. (Citadel.SSHClient.CommandFailed error 1.)**

here is my cilicon.yml
```
source: oci://ghcr.io/cirruslabs/macos-sonoma-xcode:15.3
provisioner:
  type: gitlab
  config:
    gitlabURL: https://gitlab.com
    runnerToken: <runner_token>
```

Could somebody tell me what I'm doing wrong ?

**Comment by Marcocanc** [comment]
Hi @kpolupanova, to rule out potential issues with the gitlab provisioner, could you test with a `script` type provisioner and just `echo "hello world"`?

**Comment by kpolupanova** [comment]
> 

@Marcocanc Thanks for fast answer! With scrip it's goint like this. Is it correct behaviour ? I'm newbe in CI/CD and trying to figure out how to run my pipeline by virtual machine.  
Should I create new runner for Cilicon or use existing is fine ?
<img width="524" alt="image" src="https://github.com/traderepublic/Cilicon/assets/74375888/48bb81aa-d202-4907-a189-a401632066ba">

**Comment by Marcocanc** [comment]
The behaviour with `script` seems to be fine.
Unfortunately I personally don't have a gitlab account and can't test with it. I will try to set up an account soon though.

>Should I create new runner for Cilicon or use existing is fine ?
At local machine all works correctly with runner that i use for Cilicon.

Could you explain this a bit further with more context? Do I understand it correctly, that you are able to spin up a gitlab runner on your local machine with the same runner token you were trying to use in Cilicon?

**Comment by kpolupanova** [comment]
> Could you explain this a bit further with more context? Do I understand it correctly, that you are able to spin up a gitlab runner on your local machine with the same runner token you were trying to use in Cilicon?

Sure! 

Well, I've tried set up Cilicon with new gitlab runner that was't registred at all at Host OS machine, and got the same error, then tried to set up Cilicon with existing runner token that was registred at Host OS machine. 
So I have same result in each variant and doesn't understand how i need to do it propertly.

And after all, for being sure that i don't have any problem with runners I checked that runners work correctly at Host OS machine.

**Comment by janckoch** [comment]
I have the same Problem here. I am trying to connect the cilicon runner to a self-hosted gitlab instance and i end up in an endless loop with the aforementioned error

![image](https://github.com/traderepublic/Cilicon/assets/2143327/a99bb0a8-1910-4974-96e0-d2dbb162fc75)

The cilicon.yml is
source: oci://ghcr.io/cirruslabs/macos-sonoma-xcode:15.4
provisioner:
  type: gitlab
  config:
     gitlabURL: <gitlab_url>
     runnerToken: <runner_token>

And the cilicon version is 2.2.2

**Comment by alanrickman** [comment]
I'm seeing the same behaviour as described above. My `cilicon.yml` looks like this:

```
source: oci://ghcr.io/cirruslabs/macos-sonoma-xcode:15.3
provisioner:
 type: gitlab
 config:
   gitlabURL: https://xxxx
   runnerToken: glrt-xxxxx
```

It seems like the issue was introduced in Cilicon v2.2.1. The runner successfully connects to GitLab if I downgrade to v2.2.0.

**Comment by kpolupanova** [comment]
@alanrickman thanks for your reply, I'll try with a downgrade, and hope that this issue will be fixed in future release

**Comment by Marcocanc** [comment]
@kpolupanova did downgrading fix the issue?

**Comment by kpolupanova** [comment]
> @kpolupanova did downgrading fix the issue?

@Marcocanc yes after downgrading everything works fine, but I hope this will be fixed in future updates

**Comment by roberto2lini** [comment]
Just wanted to mention I have the same issue. Downgrading from v2.2.2 to v2.2.0 works!

**Comment by phlmn** [comment]
@Marcocanc Haha, while looking into this I just sumbled upon your comment here: https://github.com/orlandos-nl/Citadel/issues/87#issuecomment-2805058673

Is there anything left to do? Let me know if I can help.

**Comment by Marcocanc** [comment]
This should be fixed with https://github.com/traderepublic/Cilicon/releases/tag/v2.2.3
Could someone please confirm?

**Comment by phlmn** [comment]
Yes, now it works :)

**Comment by Marcocanc** [comment]
Awesome!
